### PR TITLE
Save credential token locally and securely

### PR DESCRIPTION
**Pull request by yousefa00**
This PR adds a method to securely store the token containing a user's credentials for future use. Similar to the OAuth2l wrapper, the token is stored in memory rather than being stored in localstorage or in a cookie to protect from XSS and CSRF attacks.

For convenience, the token will automatically refresh 1 minute from expiry if the user remains on the page to provide a continuous user experience.

Future enhancements can include using a refresh token and implementing a backend db so that the token can persist even when a user closes the current tab (being stored in memory means the credential token is lost once the tab is closed).

### CODE REVIEW

File: web/api/backend.go [hunk 1]
```diff
@@ -83,8 +88,8 @@ func wrapperExecutor(wc WrapperCommand) string {
 // createCredentialsToken takes as input a representiation of the uplodaded uploadCredentials json body and returns a token using the
 // uploadCredentials json body as the payload.
 func createCredentialsToken(Credentials map[string]interface{}) (string, error) {
-	// Declare the expiration time of the token. Here, we have kept it as 1 day.
-	expirationTime := time.Now().Add(1440 * time.Minute)
+	// Declare the expiration time of the token. Here, we have kept it as 10 minutes.
+	expirationTime := time.Now().Add(10 * time.Minute)
```
Comment (andyrzhao) [inline web/api/backend.go:92]: Let's declare this value as a const and attach the comment to the constant.

File: web/client/src/components/form/TokenCredentials.jsx [hunk 2]
```diff
@@ -67,11 +76,33 @@ export default function TokenCredentials(props) {
...
+      <FormControlLabel
+        className="form-save-cred-toggle"
+        control={
+          <Switch
+            checked={saveCredential}
+            onChange={toggleSave}
+            color="primary"
+          />
+        }
+        label="Save credential for future use"
```
Comment (andyrzhao) [inline web/client/src/components/form/TokenCredentials.jsx:95]: credentials (for all user facing usage of this word, use the plural form to be consistent)

File: web/client/src/components/form/TokenCredentials.jsx [hunk 3]
```diff
@@ -88,6 +119,13 @@ export default function TokenCredentials(props) {
           control={<Radio color="primary" />}
           label="Text input"
         />
+        {tokenAvailable ? (
+          <FormControlLabel
+            value="saved"
+            control={<Radio color="primary" />}
+            label="Saved credential"
```
Comment (andyrzhao) [inline web/client/src/components/form/TokenCredentials.jsx:126]: credentials

File: web/client/src/components/form/TokenCredentials.jsx [hunk 4]
```diff
@@ -119,6 +157,19 @@ export default function TokenCredentials(props) {
...
           component={TextField}
         />
+      ) : credFormat === "saved" ? (
+        <TextFieldMUI
+          multiline
+          rows={15}
+          fullWidth
+          variant="outlined"
+          color="primary"
+          label="This is your saved credential"
```
Comment (andyrzhao) [inline web/client/src/components/form/TokenCredentials.jsx:167]: ditto

File: web/client/src/components/form/TokenCredentials.jsx [hunk 5]
```diff
@@ -128,7 +179,7 @@ export default function TokenCredentials(props) {
         )}
         {success && (
           <Alert variant="outlined" severity="success">
-            Credential ready!
+            Credential ready
```
Comment (andyrzhao) [inline web/client/src/components/form/TokenCredentials.jsx:182]: ditto

File: web/client/src/util/apiWrapper.js [hunk 6]
```diff
@@ -5,11 +5,20 @@ const BASE_URL =
     ? "https://yousefa-step-2020.uc.r.appspot.com/api"
```
Comment (andyrzhao) [inline web/client/src/util/apiWrapper.js:5]: Maybe add a TODO comment here for the long-term plan of the production endpoint as a reminder to update this. Also, maybe ask the intern web-dev consultants on the best way to host this app post-internship. Thanks!

Comment (yousefa00) [reply web/client/src/util/apiWrapper.js:5]: Will do, no problem!

**Pull request merged**